python/ray/_private/workers/default_worker.py

```diff
@@ -270,15 +271,35 @@
         worker_launched_time_ms=worker_launched_time_ms,
     )
 
+    stdout_fileno = sys.stdout.fileno()
+    stderr_fileno = sys.stderr.fileno()
+    # We also manually set sys.stdout and sys.stderr because that seems to
+    # have an effect on the output buffering. Without doing this, stdout
+    # and stderr are heavily buffered resulting in seemingly lost logging
+    # statements. We never want to close the stdout file descriptor, dup2 will
+    # close it when necessary and we don't want python's GC to close it.
+    sys.stdout = ray._private.utils.open_log(
+        stdout_fileno, unbuffered=True, closefd=False
+    )
+    sys.stderr = ray._private.utils.open_log(
+        stderr_fileno, unbuffered=True, closefd=False
+    )
+
     worker = ray._private.worker.global_worker
 
     # Setup log file.
-    out_file, err_file = node.get_log_file_handles(
-        get_worker_log_file_name(args.worker_type)
+    out_filepath, err_filepath = node.get_log_file_names(
+        get_worker_log_file_name(args.worker_type),
+        unique=False,  # C++ core worker process already creates the file, should use a deterministic function to get the same file path.
+        create_out=True,
+        create_err=True,
     )
-    configure_log_file(out_file, err_file)
-    worker.set_out_file(out_file)
-    worker.set_err_file(err_file)
+    worker.set_out_file(out_filepath)
+    worker.set_err_file(err_filepath)
+
+    # Disable file offset fetch if rotation enabled (since file offset doesn't make sense for rotated files).
+    if os.getenv("RAY_ROTATION_MAX_BYTES", None):
```

Comment:
Actually, 0 is also no rotation right? so we need to check whether it's > 0

Reply:
Basically we should have the same logic as

```
/*static*/ size_t RayLog::GetRayLogRotationMaxBytesOrDefault() {
  if (const char *ray_rotation_max_bytes = std::getenv("RAY_ROTATION_MAX_BYTES");
      ray_rotation_max_bytes != nullptr) {
    size_t max_size = 0;
    if (absl::SimpleAtoi(ray_rotation_max_bytes, &max_size) && max_size > 0) {
      return max_size;
    }
  }
  return std::numeric_limits<size_t>::max();
}
```

Reply:
I don't think 0 is a valid value for rotation

Reply:
added